### DISCUSSION TITLE
Segmentation fault when simulating s2p file with tabs

### DISCUSSION THREAD

**Discussion by stan3c** [ra3xdh/qucs_s#475] [Q&A] [answered]
I get "Segmentation fault" error and qucs-s crashes when starting ngspice simulation or trying to save a netlist. 
I can edit schematics-no problem, I can run simulations with qucsator-no problem. But when I select ngspice and hit F2 qucs-s crashes. ngspice path is configured properly in "Simulation" menu. ngspice itself is working without a glitch in batch mode. I also use the same ngspice with xschem and have no problem with it.
What could be the problem?

ngspice-40+, LinuxMX, qucs-s 24.1.0 compiled from latest github sources.

Edit: I may have found the answer. My schematic is a simple s2p symbol from qucs-s library with a reference to s2p file with two ports. I found another thread [#450 ](https://github.com/ra3xdh/qucs_s/discussions/450) where people say that ngspice does not support s2p files. I did not know about that. Another AC simulation of a schematic with a simple RC low-pass filter without s2p files runs in qucs-s with ngspice just fine.
I do not know whether ngspice supports s2p files, but could this be the reason? If so, what is the best simulator to run s2p simulations? I am unable to compile Xyce because Trilinos is not compiling on my machine for some reason.
BTW, Crashing the program just because it does not support an element is not an elegant way of telling it to the user.

**Comment by ra3xdh** [comment]
The segmentation fault is related to #392 This issue is partially fixed on the `current` branch. You may wait for the next release or compile Qucs-S from git to get this update. Ngspice supports s2p files since version 41. You have to compile Ngspice from source too. The s2p support in Ngspice may be limited. 

> I found another thread https://github.com/ra3xdh/qucs_s/discussions/450where people say that ngspice does not support s2p files

#450 is related only to XYCE simulator.  

> If so, what is the best simulator to run s2p simulations? 

Qucsator has native support of RF devices including touchstone files. You may use it, but keep in ind poor performance of the Qucsator on complex circuits and in the time domain. Qucsator is good for RF but not recommended as the general purpose circuit simulator.

**Reply by stan3c** [reply]
[s2p-chart.tar.gz](https://github.com/ra3xdh/qucs_s/files/14002995/s2p-chart.tar.gz)
Well, both ngspice and qucs-s on my machine are compiled from sources from github and the sources were cloned 4 days ago. So maybe the bug you are talking about is not fixed?
I attached my schematic and s2p files.

**Comment by ra3xdh** [comment]
The crash is not related to #392. The s2p file uses Tab instead of spaces as delimiter. You can replace tab by spaces as a workaround and simulation will work. I will add a fix soon.

**Comment by ra3xdh** [comment, marked as answer]
I have added a fix for crash. You may get the latest version from git.

**Comment by dwarning** [comment]
A bit off topic: I use s-parameter files to compare foundry measurements with model performances. Question is, how can I use s-parameter from different operatings points in parametric simulations only in plotting. Tried to put them in one file doesn*t work (with and without an empty line between the data blocks). Using for each file an 2-port setup? Looks a bit elaborate.

**Reply by stan3c** [reply]
You can put two files on one schematic and add 4 ports. Then you can plot on one graph for example s21 and s43.
